envs/env_4inrow.py

```diff
@@ -37,17 +37,59 @@ def render(self):
 
     def available_moves(self, player):
         moves = []
-        for i in range(self.boardW):
-            for j in reversed(range(self.boardH)):
-                if self.board[j][i] == 0:
-                    moves.append((j, i))
+        for j in range(self.boardW):
+            for i in reversed(range(self.boardH)):
+                if not self.board[i][j]:
+                    moves.append((i, j))
         return moves
 
     def is_terminal_state(self):
         for player in [self.player1, self.player2]:
             if self.available_moves(player) == [] \
-                    or self.player_status(player) != 0:
+                    or self.player_status(player):
                 return True
 
     def player_status(self, player):
-        pass
+
+        # Iteration on each cell
+        for i in reversed(range(self.boardH)):
+            for j in range(self.boardW):
+
+                # check if not zero and remember the cell symbol
+                if self.board[i][j]:
+                    symbol = self.board[i][j]
+
+                    # check four in row: up
+                    if (i-IN_ROW+1) >= 0 \
+                            and all(self.board[i-k][j] == symbol for k in range(1, IN_ROW)):
```

Comment:
Also I've notice -  
Suppose we have a board `6x7`. What happens at `(i, j) = (0, 0)` ?  
Or even at `(i, j) = (2, 2)` ?  
You might want to test these function before comitting :)

Reply:
I did not understand. What is the problem?
The truth - I did try the function before I uploaded it...

Reply:
Oh sorry, I didn't notice the first condition. That's ok then, however the second approach of scanning through the board rather than the cells individually would still be more efficient.

Reply:
That's OK

Reply:
Is this resolved?

Reply:
No, I requested changes for implementation.